modules/snooker/src/main/ts/ephox/snooker/operate/ModificationOperations.ts

```diff
@@ -10,16 +10,17 @@ type Subst = (element: SugarElement, comparator: CompElm) => SugarElement;
 // example is the location of the cursor (the row index)
 // index is the insert position (at - or after - example) (the row index)
 const insertRowAt = function (grid: Structs.RowCells[], index: number, example: number, comparator: CompElm, substitution: Subst) {
-  const before = grid.slice(0, index);
-  const after = grid.slice(index);
+  const { rows, cols } = GridRow.extractGridDetails(grid);
+  const before = rows.slice(0, index);
+  const after = rows.slice(index);
 
-  const between = GridRow.mapCells(grid[example], function (ex, c) {
-    const withinSpan = index > 0 && index < grid.length && comparator(GridRow.getCellElement(grid[index - 1], c), GridRow.getCellElement(grid[index], c));
-    const ret = withinSpan ? GridRow.getCell(grid[index], c) : Structs.elementnew(substitution(ex.element, comparator), true);
+  const between = GridRow.mapCells(rows[example], function (ex, c) {
+    const withinSpan = index > 0 && index < rows.length && comparator(GridRow.getCellElement(rows[index - 1], c), GridRow.getCellElement(rows[index], c));
+    const ret = withinSpan ? GridRow.getCell(rows[index], c) : Structs.elementnew(substitution(ex.element, comparator), true);
     return ret;
   });
 
-  return before.concat([ between ]).concat(after);
+  return cols.concat(before).concat([ between ]).concat(after);
```

Comment:
There has to be a better way than this chain of concats.

Reply:
I'm afraid there's not, unfortunately... the alternative would be to do something like this:
```
return [ ...cols, ...before, ...between, ...after ]
```
However, this from memory is [slower](https://stackoverflow.com/a/57191593/11275515) and ends up in larger bundled code.

Reply:
Joy.
But fair, this is the best we can do for now.